src/Microsoft.ML.StandardLearners/Standard/SdcaBinary.cs

```diff
@@ -1574,8 +1509,131 @@ private protected override void CheckLabel(RoleMappedData examples, out int weig
             weightSetCount = 1;
         }
 
-        protected override BinaryPredictionTransformer<TScalarPredictor> MakeTransformer(TScalarPredictor model, Schema trainSchema)
-            => new BinaryPredictionTransformer<TScalarPredictor>(Host, model, trainSchema, FeatureColumn.Name);
+        protected override BinaryPredictionTransformer<TModelParameters> MakeTransformer(TModelParameters model, Schema trainSchema)
+            => new BinaryPredictionTransformer<TModelParameters>(Host, model, trainSchema, FeatureColumn.Name);
+    }
+
+    public sealed class SdcaCalibratedBinaryTrainer :
+        SdcaBinaryTrainerBase<CalibratedModelParametersBase<LinearBinaryModelParameters, PlattCalibrator>>
+    {
+        internal const string LoadNameValue = "SDCALR";
+        internal const string UserNameValue = "Fast Linear (SA-SDCA) Logistic Regression";
+
+        /// <summary>
+        /// Configuration to training logistic regression using SDCA.
+        /// </summary>
+        public sealed class Options : BinaryArgumentBase
+        {
+        }
```

Comment:
why should this exist?

Reply:
It's for adding `sealed` on top-level trainer's `Options`. Note that `BinaryArgumentBase` is also used to derive the `Options` in `SdcaBinaryTrainer`.

---
In reply to: [256257636](https://github.com/dotnet/machinelearning/pull/2506#discussion_r256257636) [](ancestors = 256257636)